libs/expo/betterangels/src/lib/providers/featureFlag/featureFlagProvider.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useMemo, useState } from "react";
+import { gql, useQuery } from "@apollo/client";
+import { FeatureFlagContext, FeatureFlags } from "./featureFlagContext";
+import { View, Text } from "react-native";
+
+const FEATURE_FLAGS_QUERY = gql`
+    query GetFeatureFlags {
+        featureControls {
+            flags {
+                name
+                isActive
+                lastModified
+            }
+        }
+    }
+`;
+
+interface FeatureFlagProviderProps {
+    children: React.ReactNode;
+}
+
+export const FeatureFlagProvider: React.FC<FeatureFlagProviderProps> = ({ children }) => {
+    const [featureFlags, setFeatureFlags] = useState<FeatureFlags>({});
+    const { data, loading, error } = useQuery(FEATURE_FLAGS_QUERY);
```

Comment:
We can use an autogenerated hook here, so there's no need to pass in the query manually. This approach also provides a clean return type for the query. You can see an example here: [UserProvider.tsx](https://github.com/BetterAngelsLA/monorepo/blob/f6d0f9e049578d3761c8d8dd5d7dccae6780ddfa/libs/expo/betterangels/src/lib/providers/user/UserProvider.tsx#L15-L17).

Running ynx-generate within the dev container should generate the necessary types for you.